src/shared/sleep-config.c

```diff
@@ -204,15 +203,23 @@ static int swap_device_to_major_minor(const SwapEntry *swap, char **ret) {
         if (r < 0)
                 return log_debug_errno(errno, "Unable to stat %s: %m", swap->device);
 
-        swap_dev = streq(swap->type, "partition") ? sb.st_rdev : sb.st_dev;
-        if (asprintf(&major_minor, "%u:%u", major(swap_dev), minor(swap_dev)) < 0)
-                return log_oom();
+        if (streq(swap->type, "partition"))
+                swap_dev = sb.st_rdev;
```

Comment:
this really should check if S_ISBLK(st.st_mode) first, no? (and return -ENOTBLK if not true)